### PR TITLE
[ecal] Add new port

### PR DESCRIPTION
**Pull request by DownerCase**
Continues: #19802

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

Important notes:
- Port is static only on Windows due to total lack of dllexport in some components
- Core libraries are now built according to `BUILD_SHARED_LIBS` (previously shared only)
- No apps or plugins are built.
  - Not required for MVP
  - How to handle the plugins correctly was not obvious
- eCAL config files were removed
  - Defaults are used if the file cannot be found
- Value of `ECAL_INSTALL_PREFIX` cleared. The value of this define tells eCAL where to look for plugins and the config file.
  - It needs to resolve to the vcpkg install directory but:
    - It can't be an absolute path
    - It gets resolved relative to the running application
  - I currently do not see a solution to this.
- Yes, this port also installs a CMake package called `CMakeFunctions` hence the two `vcpkg_cmake_config_fixup` calls.
  - The package is provided by eCAL and is expected by users.
  - It does not appear to exist outside of eCAL despite being in a "thirdparty" folder...
- The usage text omits some targets that exist but are more utility targets for eCAL.

### CODE REVIEW

**Comment by DownerCase** [comment]
Nice, CI passed first try. 
I've tested that everything works ok via the eCAL samples for the x64-windows (+static) and x64-linux (+dynamic) triplets. Only hint anything is different from official distribution is that they all complain about not being able to find the config file (as noted in OP).
Repo: https://github.com/DownerCase/vcpkg_verifier/tree/ecal

**Comment by FrankXie05** [comment]
cc @BillyONeal

File: ports/ecal/0001-disable-app-plugins.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 4ea1fb066..351fe6c3f 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -351,8 +351,10 @@ endif(HAS_QT5)
+ # --------------------------------------------------------
+ # ecal rec addon sdk
+ # --------------------------------------------------------
++if (DBUILD_APPS)
```
Comment (dan-shaw) [inline ports/ecal/0001-disable-app-plugins.patch:9]:
```suggestion
+if (BUILD_APPS)
```

Comment (DownerCase) [reply ports/ecal/0001-disable-app-plugins.patch]: Good eye! Done: e4b7cd63c4e4c2c4951ee4ac14acf717d4620744

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**